master/internal/api_tasks.go

```diff
@@ -219,7 +219,27 @@ func (a *apiServer) GetAllocation(
 	ctx context.Context,
 	req *apiv1.GetAllocationRequest,
 ) (*apiv1.GetAllocationResponse, error) {
-	return nil, grpcutil.UnimplementedError
+	if req.AllocationId == "" {
+		return nil, status.Error(codes.InvalidArgument, "allocation ID missing")
+	}
+
+	resp := model.Allocation{}
+
+	err := db.Bun().NewRaw(`
+	SELECT allocation_id, state, slots, exit_reason, status_code
+	FROM allocations
+	WHERE allocation_id = ?
+	`, req.AllocationId).Scan(ctx, &resp)
+	if err != nil {
+		return nil, fmt.Errorf("querying allocation %s: %w", req.AllocationId, err)
+	}
+	return &apiv1.GetAllocationResponse{
+		AllocationId: string(resp.AllocationID),
+		State:        resp.State.Proto(),
+		Slots:        int32(resp.Slots),
+		ExitReason:   resp.ExitReason,
+		StatusCode:   resp.StatusCode,
```

Comment:
just realized we broke our own API style and embedded all the allocation fields in `GetAllocationResponse` instead of adding an `Allocation` type that is reusable if we add a `List..` later. (see `GetTrialsResponse`). Thoughts?